src/op_analytics/cli/subcommands/pulls/defillama.py

```diff
@@ -88,60 +83,111 @@ def process_breakdown_stables(data):
         metadata[key] = data.get(key)
 
-    result = pl.DataFrame(rows, infer_schema_length=len(rows)).with_columns(
-        id=pl.lit(metadata["id"]),
-        name=pl.lit(metadata["name"]),
-        symbol=pl.lit(metadata["symbol"]),
+    result = (
+        pl.DataFrame(rows, infer_schema_length=len(rows)).with_columns(
+            id=pl.lit(metadata["id"]),
+            name=pl.lit(metadata["name"]),
+            symbol=pl.lit(metadata["symbol"]),
+        )
+        if rows
+        else pl.DataFrame()
     )
 
     return result, metadata
 
 
-def pull_stables():
-    """Pull stablecoin data from DeFiLlama.
+def pull_stables(
+    stablecoin_ids: Optional[List[str]] = None, days: int = 30
+) -> Dict[str, pl.DataFrame]:
+    """
+    Pulls and processes stablecoin data from DeFiLlama.
+
+    Args:
+        stablecoin_ids: List of stablecoin IDs to process. Defaults to None (process all).
+        days: Number of days of data to retrieve. Defaults to 30.
 
-    - Fetch the stablecoins summary endpoint.
-    - For each stablecoin, fetch the detailed breakdown data.
-    - Write all results to BigQuery.
+    Returns:
+        A dictionary containing metadata and breakdown DataFrames.
     """
     session = new_session()
     summary = get_data(session, SUMMARY_ENDPOINT)
 
+    if "peggedAssets" not in summary:
+        raise KeyError("The 'peggedAssets' key is missing from the summary data.")
+
     urls = {}
     for stablecoin in summary["peggedAssets"]:
         stablecoin_id = stablecoin["id"]
-        urls[stablecoin_id] = BREAKDOWN_ENDPOINT.format(id=stablecoin_id)
+        if stablecoin_ids is None or stablecoin_id in stablecoin_ids:
+            urls[stablecoin_id] = BREAKDOWN_ENDPOINT.format(id=stablecoin_id)
+    if not urls:
+        raise ValueError("No valid stablecoin IDs provided.")
 
-    stablecoin_data = run_concurrently(lambda x: get_data(session, x), urls, max_workers=4)
+    stablecoin_data = run_concurrently(
+        lambda x: get_data(session, x), urls, max_workers=4
+    )
 
-    breakdown_dfs = []
-    metadata_rows = []
-    for data in stablecoin_data.values():
-        breakdown_df, metadata = process_breakdown_stables(data)
-        breakdown_dfs.append(breakdown_df)
-        metadata_rows.append(metadata)
+    breakdown_dfs: List[pl.DataFrame] = []
+    metadata_rows: List[Dict[str, Optional[str]]] = []
 
-    breakdown_df = pl.concat(breakdown_dfs, how="diagonal_relaxed")
-    metadata_df = pl.DataFrame(metadata_rows, infer_schema_length=len(metadata_rows))
+    for data in stablecoin_data.values():
+        breakdown_df, metadata = process_breakdown_stables(data, days=days)
+        if not breakdown_df.is_empty():
+            breakdown_dfs.append(breakdown_df)
+            if metadata:
+                metadata_rows.append(metadata)
+
+    breakdown_df = (
+        pl.concat(breakdown_dfs, how="diagonal_relaxed")
+        if breakdown_dfs
+        else pl.DataFrame()
+    )
+    metadata_df = (
```

Comment:
this is addressing my own code that I wrote before. Maybe it would be good to have `process_breakdown_stables` return a metadata dataframe instead of a metadata list of dicts. That way it would be more uniform and we could use pl.concat() jus like we do for the breakdown df.